src/index.js

```diff
@@ -4,12 +4,33 @@ import { IntlProvider, addLocaleData } from 'react-intl';
 import en from 'react-intl/locale-data/en';
 import './index.css';
 import CssBaseline from '@material-ui/core/CssBaseline';
+import { Provider } from 'react-redux';
+import { createStore, combineReducers } from 'redux';
+import throttle from 'lodash/throttle';
+import { composeWithDevTools } from 'redux-devtools-extension';
+import appReducer from './reducers/appReducer';
+import userReducer from './reducers/userReducer';
 import App from './App';
 import messagesEn from './translations/en.json';
 import * as serviceWorker from './serviceWorker';
-import { saveStateToLocalStorage, hydrateStateWithLocalStorage } from './utils';
+import { loadState, saveState } from './localStorage';
 
-const props = { saveStateToLocalStorage, hydrateStateWithLocalStorage };
+const persistedState = loadState();
+const store = createStore(
+  combineReducers({
+    appReducer,
+    userReducer
+  }),
+  persistedState,
+  composeWithDevTools()
+);
+
+// https://egghead.io/lessons/javascript-redux-persisting-the-state-to-the-local-storage
+store.subscribe(
```

Comment:
Should we really be saving the state to the local storage every single time the store's states is changed? should this be instead done on window close even or the root app component's life cycle ?

Reply:
I don't know what the best way of doing this is. At least the throttle function prevents it from saving it all the time. 

The previous implementation had it connected to the `componentWillUnmount` method of the App component. Then again I don't know how reliable it is, and the JSON.stringify and parse methods are quite heavy so there might be situations where the saving does not complete properly if we use that approach.

I assumed that the author of redux would know best 😅

Reply:
well of course its a valid recommendation, however since the JSON.stringify and parse methods are heavy that would be in favor of not calling them all the time. One compromise would be to save user's state every time the user navigate between different views, this way we call the function more often than just on componentWillUnmount but still not every time we update the store.

Reply:
@mansour-ahmed I took out the throttle from the subscription so it looked like this:
```
store.subscribe(() => {
  saveState(store.getState());
});
``` 
With one year amount of transactions, I compared the performance with and without it. Having the Firefox's performance profiler on, and clicking between landing and info page, there was no difference in re-render times. With both settings, one click took around 300 - 500 ms on average. So it seems that there is no big impact on saving it on each time